python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py

```diff
@@ -244,3 +244,91 @@ def test_avro_compression(rows, codec):
     got_df = cudf.read_avro(buffer)
 
     assert_eq(expected_df, got_df)
+
+
+avro_logical_type_params = [
+    # (avro logical type, avro primitive type, cudf expected dtype)
+    ("date", "int", "datetime64[s]"),
+]
+
+
+@pytest.mark.parametrize(
+    "logical_type, primitive_type, expected_dtype", avro_logical_type_params
+)
+@pytest.mark.parametrize("namespace", [None, "root_ns"])
+@pytest.mark.parametrize("nullable", [True, False])
+@pytest.mark.parametrize("prepend_null", [True, False])
+def test_can_detect_dtypes_from_avro_logical_type(
+    logical_type,
+    primitive_type,
+    expected_dtype,
+    namespace,
+    nullable,
+    prepend_null,
+):
+    avro_type = [{"logicalType": logical_type, "type": primitive_type}]
+    if nullable:
+        if prepend_null:
+            avro_type.insert(0, "null")
+        else:
+            avro_type.append("null")
+
+    schema = fastavro.parse_schema(
+        {
+            "type": "record",
+            "name": "test",
+            "namespace": namespace,
+            "fields": [{"name": "prop", "type": avro_type}],
+        }
+    )
+
+    actual = cudf_from_avro_util(schema, [])
+
+    expected = cudf.DataFrame(
+        {"prop": cudf.Series(None, None, expected_dtype)}
+    )
+
+    assert_eq(expected, actual)
+
+
+def get_days_from_epoch(date: datetime.date) -> int:
+    return (date - datetime.date(1970, 1, 1)).days
+
+
+def test_can_parse_avro_date_logical_type():
+
+    schema = fastavro.parse_schema(
+        {
+            "type": "record",
+            "name": "test",
+            "fields": [
+                {
+                    "name": "o_date",
+                    "type": {
+                        "logicalType": "date",
+                        "type": "int",
+                    },
+                }
+            ],
+        }
+    )
+
+    dates = [
+        datetime.date(1970, 1, 1),
+        datetime.date(1970, 1, 2),
+        datetime.date(1981, 10, 25),
+        datetime.date(2012, 5, 18),
+        datetime.date(2019, 9, 3),
+    ]
```

Comment:
Would maybe be good to have some nulls mixed into a variant of this test as well to ensure that we properly handle the situations where we get a `UNION` type with nulls and dates.

Reply:
Just pushed a change that rewrites the test to test nulls where applicable.  Can you take a look?  Thanks!